client/src/app/pages/search/components/SearchMenu.tsx

```diff
@@ -0,0 +1,314 @@
+import React from "react";
+import {
+  Menu,
+  MenuContent,
+  MenuItem,
+  MenuList,
+  Popper,
+  SearchInput,
+} from "@patternfly/react-core";
+import { FILTER_TEXT_CATEGORY_KEY } from "@app/Constants";
+import { VulnerabilitySearchContext } from "@app/pages/vulnerability-list/vulnerability-context";
+import { SbomSearchContext } from "@app/pages/sbom-list/sbom-context";
+import { Label } from "@patternfly/react-core";
+import { AdvisorySearchContext } from "@app/pages/advisory-list/advisory-context";
+import { PackageSearchContext } from "@app/pages/package-list/package-context";
+
+export interface IEntity {
+  id: string;
+  title: string | null;
+  description?: string | null;
+  navLink: string;
+  type: string;
+  typeColor:
+    | "purple"
+    | "orange"
+    | "blue"
+    | "cyan"
+    | "green"
+    | "red"
+    | "grey"
+    | "gold"
+    | undefined;
+}
+
+function useAllEntities() {
+  const [entityList, setEntityList] = React.useState<IEntity[]>([]);
+
+  const {
+    tableControls: { currentPageItems: advisories },
+  } = React.useContext(AdvisorySearchContext);
```

Comment:
This is extracting the list of advisories of the SearchPageContext but I think we should change it:

- Let's say the Backend has one thousand `Advisories`
- The `AdvisorySearchContext` contains the information of the data to be rendered in the table (10 elements)
- If extract data from `AdvisorySearchContext` here then it means the dropdown will only contain elements from the current table. But I think what should be done is to search in the backend according to what the user types regarless of what the values of `AdvisorySearchContext` are.

So this block of code could be replaced by:

```typescript
function useAllEntities(filterText: string) {
  const [entityList, setEntityList] = React.useState<IEntity[]>([]);

  const {
    result: { data: advisories },
  } = useFetchAdvisories({
    filters: [
      { field: FILTER_TEXT_CATEGORY_KEY, operator: "~", value: filterText },
    ],
    page: { pageNumber: 1, itemsPerPage: 10 },
  });
```

In the code I am sharing the function `useAllEntities` receives a param `filterText` which represents what the user types in the input text, then that string is send to the backend to do a filter among the hypothetical thousand Advisories in the database

Same logic can be applied to the other 3 entities of this function (Packages, Sboms, Vulnerabilities)